dan_layer/core/src/workers/hotstuff_waiter.rs

```diff
@@ -388,63 +407,90 @@ where
         let b_one = tx.get_node(&b_two.justify().local_node_hash()).map_err(|e| e.into())?;
 
         let (_b_lock, b_lock_height) = tx.get_locked_node_hash_and_height(shard).map_err(|e| e.into())?;
-        if b_one.height().0 > b_lock_height.0 {
+        if b_one.height() > b_lock_height {
             info!(target: LOG_TARGET, "Updating locked node to: {:?}", b_one.hash());
             tx.set_locked(shard, *b_one.hash(), b_one.height())
                 .map_err(|e| e.into())?;
         }
 
         if node.justify().payload_height() == NodeHeight(2) {
-            // decide
-            debug!(target: LOG_TARGET, "Deciding on payload: {:?}", node.payload_id());
-            let results = self.on_commit(node, shard, &mut tx)?;
-            self.publish_event(HotStuffEvent::OnCommit(node_hash, results));
+            match finalize_result {
+                Some(result) => {
+                    self.publish_result_event(node.justify().payload_id(), result);
+                    let changes = extract_changes(node.payload_id(), result)?;
+                    info!(
+                        target: LOG_TARGET,
+                        "payload changeset: {}",
+                        serde_json::to_string(&changes).unwrap()
+                    );
+                    self.on_commit(node, &changes, &mut tx)?;
+                },
+                None => {
+                    warn!(
+                        target: LOG_TARGET,
+                        "No finalize result for payload {}. Not enough votes?",
```

Comment:
This is actually an error. It should never get to this point without a finalize result, so there is another cause. (I'm not sure what it is)
```suggestion
                        "No finalize result for payload {}. ",
```

Reply:
Thanks, wasn't sure